src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -185,6 +185,33 @@ export class TestDataProvider implements ITestDataProvider {
     );
   }
 
+  public async updateTestTypeDetails(testTypes: models.TestType[], testTypeParams: models.TestTypeParams): Promise<models.TestType[]> {
+    return await Promise.all(
+      testTypes.map(async (testType) => {
+        const { testTypeId } =  testType
+        const fields = "defaultTestCode,linkedTestCode,testTypeClassification,name,testTypeName"
+        const {
+          defaultTestCode,
+          linkedTestCode,
+          testTypeClassification,
+          name,
+          testTypeName
+        } = await this.testResultsDAO?.getTestCodesAndClassificationFromTestTypes(testTypeId, testTypeParams, fields)
+        return {
+          ...testType,
+          testTypeClassification,
+          testCode:
+            testTypes.length > 1 && linkedTestCode
+              ? linkedTestCode
+              : defaultTestCode,
```

Comment:
```suggestion
          testCode: (testTypes.length > 1 && linkedTestCode) ?? defaultTestCode,
```
This produces the same output.

Reply:
This method is essentially a combination of 3 methods that used to be called. I wanted to keep the code the same to keep it consistent with the other methods still called by the `POST` endpoint, to possibly simplify refactoring and readability in the future. Great suggestion though!